### DISCUSSION TITLE
CCMenu2 pre-releases

### DISCUSSION THREAD

**Discussion by erikdoe** [erikdoe/ccmenu2#4] [General]
This is a **pre-release** for CCMenu2. It will not be published via the AppStore. There is no update check in the versions released on GitHub. You will **not** be notified within the app when a new pre-release is published.

Please join the [GitHub discussion](https://github.com/erikdoe/ccmenu2/discussions/3) for this release to provide feedback. If you find a bug you can open an issue but please do not open issues for missing or incomplete features at this stage.

<hr /><em>This discussion was created from the release <a href='https://github.com/erikdoe/ccmenu2/releases/tag/v20.1'>CCMenu 20.1</a>.</em>

**Comment by ryanong** [comment]
Giving full control to ccmenu is not something my employer wishes to do, is there anyway to view an actions status without full permissions?

**Comment by erikdoe** [comment]
Good question! I'm still trying to figure this out. The documentation isn't all that clear: https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps

In any case, GitHub seem to steer application authors towards GitHub Apps (CCMenu is currently an OAuth App), but as far as I understand GitHub Apps have to be "installed" by an admin of the org. For users of CCMenu this would mean that after installing CCMenu on their computer they would then have to ask the admin of the org to "install" the GitHub App in the organisation. Haven't understood yet whether those access rights would be more fine-grained. In additions, there's added complexity because tokens for GitHub Apps must be refreshed every few hours: https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-user-access-token-for-a-github-app

Users can create OAuth tokens in the web UI, too (https://github.com/settings/tokens/new), but I think that doesn't add anything beyond what the current flow allows for.

So, I hear you, and I would like to use more restrictive scope myself, but I haven't quite found out how to proceed.

**Comment by erikdoe** [comment]
There's now a new pre-release that has support for GitHub fine-grained personal access tokens. If it's still relevant. Have a look: https://github.com/ccmenu/ccmenu2/releases/tag/v28.0d3
